ang/civicase/case/services/add-case.service.js

```diff
@@ -0,0 +1,85 @@
+(function (_, angular, checkPerm, loadForm, getCrmUrl) {
+  var module = angular.module('civicase');
+
+  module.service('AddCase', AddCaseService);
+
+  /**
+   * Add Case Service
+   *
+   * @param {object} $window the window service.
+   * @param {string} CaseCategoryWebformSettings service to fetch case category webform settings
+   */
+  function AddCaseService ($window, CaseCategoryWebformSettings) {
+    this.clickHandler = clickHandler;
+    this.isVisible = isVisible;
+
+    /**
+     * Displays a form to add a new case. If a custom "Add Case" webform url has been configured,
+     * it will redirect to it. Otherwise it will open a CRM form popup to add a new case.
+     *
+     * @param {string} caseTypeCategory case type category
+     * @param {string} contactId contact id
+     * @param {Function} callback callback function
+     */
+    function clickHandler (caseTypeCategory, contactId, callback) {
```

Comment:
Can we avoid sending multiple parameters and send an object instead? There are 3 advantages when we do this:

1 - We don't need to remember the order of the parameters.
2 - When we introduce a new parameter, but need to skip previous ones we don't need to send defaults (ex: `clickHandler('case', null, null, true)`).
3 - Parameters can be understood better. Reading `clickHandler('case', '5')` is a bit confusing for a second. What does `case` mean? what does `5` mean?
```js
clickHandler({
  caseTypeCategory: 'case',
  contactId: '5'
});
```
This is more verbose, but explains it better.

If we need to define the parameters we can use `@typedef` [docs](https://jsdoc.app/tags-typedef.html), [example](https://github.com/compucorp/uk.co.compucorp.civicase/blob/master/ang/civicase-base/providers/activity-forms.provider.js#L69-L73).

Can we do this as well for the `isVisible` ?

Reply:
`isVisible` does not have any parameters.